### PR TITLE
feat(telemetry): debug environment variable

### PR DESCRIPTION
**Pull request by avara1986**
## Checklist
Add `DD_TELEMETRY_DEBUG` environment variable. This flag enables Telemetry payload debug mode.

We will need this refactor in [Telemetry Metrics](https://github.com/DataDog/dd-trace-py/pull/4943)

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.

### CODE REVIEW

File: ddtrace/internal/telemetry/writer.py [hunk 1]
```diff
@@ -54,11 +55,14 @@ def __init__(self, agent_url=None):
...
         self._lock = forksafe.Lock()  # type: forksafe.ResetObject
         self._forked = False  # type: bool
+        # Debug flag that enables payload debug mode.
+        self._debug = asbool(os.environ.get("DD_TELEMETRY_DEBUG", "false"))
         forksafe.register(self._fork_writer)
 
         self._headers = {
             "Content-type": "application/json",
             "DD-Telemetry-API-Version": "v1",
+            "DD-Telemetry-Debug-Enabled": str(self._debug).lower(),
```
Comment (mabdinur) [inline ddtrace/internal/telemetry/writer.py:65]: Can we make `self._debug` a string? Do we plan to use it as a boolean?

Comment (emmettbutler) [reply ddtrace/internal/telemetry/writer.py:65]: @mabdinur I'm curious to understand why it's not acceptable for `self._debug` to be a boolean.

**Review by emmettbutler — COMMENTED** [review]
I don't have anything to add beyond @mabdinur's comments!

File: ddtrace/internal/telemetry/writer.py (hunk 1 above)
Comment (mabdinur) [reply ddtrace/internal/telemetry/writer.py:65]: It can be a boolean but right now it's only used as a string

Comment (avara1986) [reply ddtrace/internal/telemetry/writer.py:65]: We need a bool value type here -> https://github.com/DataDog/dd-trace-py/pull/5020/files#diff-0cb9315daf5ed5208de657467cc10a6b7dda616a161c9b5b1f8cff83c95894deR261

**Review by juanjux — APPROVED** [review]

**Review by christophe-papazian — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**